contracts/interchain-token/InterchainToken.sol

```diff
@@ -97,7 +98,7 @@ contract InterchainToken is InterchainTokenStandard, ERC20Permit, Minter, IInter
          * Also add the provided address as a minter. If `address(0)` was provided,
          * add it as a minter to allow anyone to easily check that no custom minter was set.
          */
-        _addMinter(interchainTokenService_);
+        _addMinter(IInterchainTokenService(interchainTokenService_).tokenManagerAddress(tokenId_));
```

Comment:
feels like a weird re-entrancy to call ITS back during deployment. better to pass in the token manager, or have ITS call `transferMintership` after deployment. The latter avoids needing to change the token bytecode, and keeps the logic isolated in ITS